addons/account_payment_term/models/account_payment_term.py

```diff
@@ -47,7 +47,10 @@ def _get_due_date(self, date_ref):
             # prevent calculation from moving beyond the end of the next month (e.g., early March instead of end February)
             if self.days_next_month in {'29', '30', '31'}:
                 # We get the min of the days the user enter in the payment term and the last day of the next month
-                days_next_month = relativedelta(days=min(int(self.days_next_month), (date_end_of_month + relativedelta(month=2)).day))
+                days_next_month = relativedelta(days=0)
+                if int(self.days_next_month) < date_end_of_month.day:
```

Comment:
if we really want that behavior (need to see with po) then a comment would be great